engine/runtime-language-epb/src/main/java/org/enso/interpreter/epb/EpbContext.java

```diff
@@ -32,20 +36,65 @@ public EpbContext(TruffleLanguage.Env env) {
   }
 
   /**
-   * Initializes the context. No-op in the inner context. Spawns the inner context if called from
-   * the outer context.
+   * Initializes the context.No-op in the inner context. Spawns the inner context if called from the
+   * outer context. Shielded against double initialization.
+   *
+   * @param preInitializeLanguages comma separated list of languages to immediately initialize
    */
-  public void initialize() {
+  public void initialize(String preInitializeLanguages) {
     if (!isInner) {
-      innerContext =
-          new GuardedTruffleContext(
-              env.newInnerContextBuilder()
-                  .initializeCreatorContext(true)
-                  .inheritAllAccess(true)
-                  .config(INNER_OPTION, "yes")
-                  .build(),
-              true);
+      if (innerContext == null) {
+        innerContext =
+            new GuardedTruffleContext(
+                env.newInnerContextBuilder()
+                    .initializeCreatorContext(true)
+                    .inheritAllAccess(true)
+                    .config(INNER_OPTION, "yes")
+                    .build(),
+                true);
+      }
+      initializeLanguages(env, innerContext, preInitializeLanguages);
+    }
+  }
+
+  private static void initializeLanguages(
+      TruffleLanguage.Env environment, GuardedTruffleContext innerContext, String langs) {
+    if (langs == null || langs.isEmpty()) {
+      return;
+    }
+    var log = environment.getLogger(EpbContext.class);
+    log.log(Level.FINE, "Initializing languages {0}", langs);
+    var cdl = new CountDownLatch(1);
```

Comment:
Shouldn't this be constructed with the number of initialized languages? Maybe refactor the method signature to take `List<String> langs` instead of just `String`?

Reply:
The goal of the latch is to wait while the JavaScript (or Python or R) lock is taken. Once it is we can go on.